data/index.ts

```diff
@@ -0,0 +1,12 @@
+/**
+ * We need this file so we can reliably determine the location of the data directory,
+ * since everything in node is relative to either whichever file being executed or
+ * the working directory, both of which are unreliable
+ */
+
+import * as path from "path";
+import { fileURLToPath } from "url";
+
+export const DATA_DIR_PATH = path.dirname(
+    path.resolve(fileURLToPath(import.meta.url)),
+);
```

Comment:
what's the reason we need a `path.resolve` here? do we need it at all?

Reply:
We need this so we can import the `DATA_DIR_PATH` from anywhere, even when we just run some script (e.g. `init-db.ts` from some arbitrary working directory.